Source/WebKit/WebProcess/WebPage/FindController.cpp

```diff
@@ -237,6 +238,13 @@ void FindController::findString(const String& string, OptionSet<FindOptions> opt
 #if ENABLE(PDFKIT_PLUGIN)
     auto* pluginView = mainFramePlugIn();
 #endif
+    Page* corePage = m_webPage->corePage();
+
+    if (corePage->settings().liveTextForFindInPageEnabled()) {
+        Document* document = m_webPage->mainWebFrame().coreFrame()->document();
+        if (document)
+            corePage->imageAnalysisQueue().enqueueAllImages(*document, { }, { });
+    }
```

Comment:
Can we move this logic into `Page::findTextMatches` or another helper method to be called from `Page::findTextMatches`?

As it stands, we have a couple of places where `Page::findTextMatches` is used to retrieve results. One for WebKit-specific API/SPI, one for AppKit (`NSTextFinder`), and one for UIKit (`UIFindInteraction`). This would only cover the first case.

Reply:
Once you move the code into `findTextMatches`, I would also write this as something like:

```
if (settings().liveTextForFindInPageEnabled()) {
    if (RefPtr mainDocument = m_mainFrame->document())
        imageAnalysisQueue().enqueueAllImages(*mainDocument, { }, { });
}
```

On a related note, is this only invoked once at the beginning of the Find in Page invocation? (We should either ensure that this new logic is inside an entry point that's only called once at the beginning in all three scenarios, or we should adjust `enqueueAllImages` to be idempotent).

Reply:
> Can we move this logic into `Page::findTextMatches` or another helper method to be called from `Page::findTextMatches`?

Unless I'm mistaken, the case that my original patch covers is no longer covered if I move the code to `Page::findTextMatches`. From `FindController::findString`, `findTextMatches` is only called `if (selectedFrame->selection().selectionBounds().isEmpty())`

Reply:
We talked about this offline, and have decided to try "another helper method to be called from `Page::findTextMatches`", and additionally call the helper from `Page::findString`.

The difference between the WebKit-specific case and the AppKit/UIKit cases, is that WebKit only searches for the next match following the current selection, whereas AppKit/UIKit expect all results up front.

`findPlainText` in `TextIterator` is the common point between the codepaths, but not sure that the logic should live in `TextIterator`, hence the helper method approach.